x-pack/osquerybeat/internal/osqdcli/client.go

```diff
@@ -121,22 +138,31 @@ func (c *Client) Connect(ctx context.Context) error {
 
 func (c *Client) reconnect(ctx context.Context) error {
 	c.close()
+	cli, err := c.connectWithRetry(ctx, c.timeout)
+	if err != nil {
+		return err
+	}
+	c.cli = cli
+	return nil
+}
 
+func (c *Client) connectWithRetry(ctx context.Context, timeout time.Duration) (cli *osquery.ExtensionManagerClient, err error) {
 	r := retry{
 		maxRetry:  c.connectRetries,
 		retryWait: retryWait,
 		log:       c.log.With("context", "osquery client connect"),
 	}
 
-	return r.Run(ctx, func(ctx context.Context) error {
-		cli, err := osquery.NewClient(c.socketPath, c.timeout)
-		if err != nil {
-			r.log.Warnf("failed to connect, reconnect might be attempted, err: %v", err)
-			return err
+	err = r.Run(ctx, func(ctx context.Context) error {
```

Comment:
```suggestion
	err = r.Run(ctx, func(_ context.Context) error {
```
to indicate that the `fn` doesn't actually use the context.

Reply:
Done